clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/ReplacerTest.java

```diff
@@ -1284,7 +1287,7 @@ void findCronJobDockerImages() {
             artifact -> {
               assertThat(artifact).isNotNull();
               assertThat(artifact.getType()).isEqualTo("docker/image");
-              assertThat(artifact.getName()).isEqualTo("gcr.io/my-repository/my-image:my-tag");
+              assertThat(artifact.getName()).isEqualTo("gcr.io/my-repository/my-image");
```

Comment:
this is the side effect of using dockerImages in this test whose repercussions I'm not sure of.  The reference still contains the tag which seems like how it should be to me atleast.

Reply:
I need to stare at it some more...like, maybe this change ends up in an initial commit with a good explanation...and then we proceed with the other changes...

Reply:
Well, my digging hasn't been altogether satisfying.  The relevant difference between `dockerImage()` and `cronJobDockerImage()` (and really DOCKER_IMAGE vs. CRON_JOB_DOCKER_IMAGE) for this test is the `nameFromReference` member since path/findPath are I believe equivalent between the two.  Replacer.getArtifacts transforms the artifact name by using `nameFromReference`....only DOCKER_IMAGE specifies it to some code that removes the tag, and since CRON_JOB_DOCKER_IMAGE doesn't specify it, the name is the entire reference (which in this case includes the tag).

There's one other existing Replacer that operates on `KubernetesArtifactType.DockerImage`: POD_DOCKER_IMAGE.  It (like CRON_JOB_DOCKER_IMAGE) doesn't specify `nameFromReference`.

Before https://github.com/spinnaker/clouddriver/pull/5554, images in cron jobs were handled by DOCKER_IMAGE so I figure it's OK to go back to that behavior.  It does make me wonder whether POD_DOCKER_IMAGE is correct though, or whether we'd be better off if it has the same `nameFromReference` as DOCKER_IMAGE.

It seems the only place this ends up getting used outside of tests is via KubernetesHandler.listArtifacts, which is only called by KubernetesManifestContainerBuilder.buildManifest...and on up the chain to KubernetesJobProvider.getKubernetesJob (which doesn't seem relevant here since it's specific to jobs, not cron jobs) and ManifestController.getForAccountLocationAndName which orca calls, and the resulting artifacts make it into the pipeline execution context.....so changing this could potentially break SpEL expressions....

I suppose this is all something to ponder for the future.  This PR seems like progress.  It fixes artifact binding for cron jobs and I guess as a side effect, restores the naming of cron jobs back to what 1.27 did.